include/ua_plugin_securitypolicy.h

```diff
@@ -0,0 +1,424 @@
+                                   UA_ByteString *data);
+
+    /**
+     * \brief Decrypts the given ciphertext using a symmetric algorithm and key.
+     *
+     * \param securityContext the SecurityContext which contains information about the keys needed to decrypt the message.
+     * \param data the data to decrypt. The decryption is done in place.
+     */
+    UA_StatusCode (*const decrypt)(const UA_SecurityPolicy *securityPolicy,
+                                   const void *channelContext,
+                                   UA_ByteString *data);
+
+    /**
+     * \brief Pseudo random function that is used to generate the symmetric keys.
+     *
+     * For information on what parameters this function receives in what situation,
+     * refer to the OPC UA specification 1.03 Part6 Table 33
+     *
+     * \param secret
+     * \param seed
+     * \param out an output to write the data to. The length defines the maximum
+     *            number of output bytes that are produced.
+     */
+    UA_StatusCode (*const generateKey)(const UA_SecurityPolicy *securityPolicy,
+                                       const UA_ByteString *secret,
+                                       const UA_ByteString *seed,
+                                       UA_ByteString *out);
+    /**
+     * \brief Random generator for generating nonces.
+     * 
+     * Generates a nonce.
+     * 
+     * \param securityPolicy the securityPolicy this function is invoked on. Example: myPolicy->generateNonce(myPolicy, &outBuff);
+     * \param out pointer to a buffer to store the nonce in. Needs to be allocated by the caller.
+     *            The buffer is filled with random data.
+     */
+    UA_StatusCode (*const generateNonce)(const UA_SecurityPolicy *securityPolicy,
+                                         const void *endpointContext,
+                                         UA_ByteString *out);
+
+    /**
+     * \brief Calculates the padding size for a message with the specified amount of bytes.
+     *
+     * \param securityPolicy the securityPolicy the function is invoked on.
+     * \param bytesToWrite the size of the payload plus the sequence header, since both need to be encoded
+     * \param paddingSize out parameter. Will contain the paddingSize byte.
+     * \param extraPaddingSize out parameter. Will contain the extraPaddingSize. If no extra padding is needed, this is 0.
+     * \return the total padding size consisting of high and low bytes.
+     */
+    UA_UInt16 (*const calculatePadding)(const UA_SecurityPolicy *securityPolicy,
+                                        size_t bytesToWrite,
+                                        UA_Byte *paddingSize,
+                                        UA_Byte *extraPaddingSize);
+
+    const UA_SecurityPolicySigningModule signingModule;
+
+    const size_t signingKeyLength;
+    const size_t encryptingKeyLength;
+    const size_t encryptingBlockSize;
+} UA_SecurityPolicySymmetricModule;

```

Comment:
I just checked the encrypt/decrypt/calculatePadding functions of the two modules and they differ in their arguments, since the symmetric and asymmetric version use different contexts. I think it would be ok if we just unify them and live with the extra parameter that might not be used. So asymmetric and symmetric modules both have the same signature:
```
UA_StatusCode (*const encrypt)(const UA_SecurityPolicy *securityPolicy,
                                   const void *endpointContext,
                                   const void *channelContext,
                                   const UA_ByteString *data);
```